x-pack/plugins/cloud_security_posture/public/pages/vulnerabilities/vulnerabilities.tsx

```diff
@@ -49,7 +62,7 @@ export const Vulnerabilities = () => {
     return <ErrorCallout error={error as Error} />;
   }
   if (isLoading || !data) {
-    return <EuiLoadingSpinner />;
+    return <Loader />;
```

Comment:
1. in case of missing data or an error (no data if there is an error) we will show an infinite loader. since this PR is about table and flyout fixes i would fix that issue as well. error states should be handled differently from loading states.

2. is it possible to handle loaders and errors by the `CloudPosturePage` component? it makes all of our pages have the same loading and error components by default (but we can pass a custom render if we need) plus also handling the subscription state.
All it takes is wrapping this component with the `<CloudPosturePage>` and passing the query to it.

Reply:
Great catch. I agree they should be handled separately. This page is already wrapped CloudPosturePage, but that itself doesn't really guarantee that the data won't be null due to edge cases errors, so I'm reusing the components from the CloudPosturePage